tfhe/src/test_user_docs.rs

```diff
@@ -1,102 +1,109 @@
+    );
 
-//FINE GRAINED API
-doctest!(
-    "../docs/fine_grained_api/quick_start.md",
-    fine_grained_api_quick_start
-);
-// fine_grained_api/Boolean
-doctest!(
-    "../docs/fine_grained_api/Boolean/operations.md",
-    booleans_operations
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/parameters.md",
-    booleans_parameters
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/serialization.md",
-    booleans_serialization
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/readme.md",
-    booleans_tutorial
-);
+    // fine_grained_api/shortint
+    doctest!(
+        "../docs/fine_grained_api/shortint/operations.md",
+        shortint_operations
+    );
 
-// fine_grained_api/shortint
-doctest!(
-    "../docs/fine_grained_api/shortint/operations.md",
-    shortint_operations
-);
+    doctest!(
+        "../docs/fine_grained_api/shortint/parameters.md",
+        shortint_parameters
+    );
+    doctest!(
+        "../docs/fine_grained_api/shortint/serialization.md",
+        shortint_serialization
+    );
+    doctest!(
+        "../docs/fine_grained_api/shortint/readme.md",
+        shortint_tutorial
+    );
 
-doctest!(
-    "../docs/fine_grained_api/shortint/parameters.md",
-    shortint_parameters
-);
-doctest!(
-    "../docs/fine_grained_api/shortint/serialization.md",
-    shortint_serialization
-);
-doctest!(
-    "../docs/fine_grained_api/shortint/readme.md",
-    shortint_tutorial
-);
+    // fine_grained_api/integer
+    doctest!(
+        "../docs/fine_grained_api/integer/operations.md",
+        integer_operations
+    );
+    doctest!(
+        "../docs/fine_grained_api/integer/serialization.md",
+        integer_serialization_tuto
+    );
+    doctest!(
+        "../docs/fine_grained_api/integer/readme.md",
+        integer_first_circuit
+    );
 
-// fine_grained_api/integer
-doctest!(
-    "../docs/fine_grained_api/integer/operations.md",
-    integer_operations
-);
-doctest!(
-    "../docs/fine_grained_api/integer/serialization.md",
-    integer_serialization_tuto
-);
-doctest!(
-    "../docs/fine_grained_api/integer/readme.md",
-    integer_first_circuit
-);
+    // core_crypto
+    doctest!(
+        "../docs/core_crypto/presentation.md",
+        core_crypto_presentation
+    );
+    doctest!("../docs/core_crypto/tutorial.md", core_crypto_tutorial);
 
-// core_crypto
-doctest!(
-    "../docs/core_crypto/presentation.md",
-    core_crypto_presentation
-);
-doctest!("../docs/core_crypto/tutorial.md", core_crypto_tutorial);
+    // Tutorials
+    doctest!(
+        "../docs/tutorials/ascii_fhe_string.md",
+        ascii_fhe_string_tutorial
+    );
+    doctest!("../docs/tutorials/parity_bit.md", parity_bit_tutorial);
+}
 
-// Tutorials
-doctest!(
-    "../docs/tutorials/ascii_fhe_string.md",
-    ascii_fhe_string_tutorial
-);
-doctest!("../docs/tutorials/parity_bit.md", parity_bit_tutorial);
+#[cfg(feature = "gpu")]
+fn test_gpu_doc() {
+    doctest!("../docs/how_to/run_on_gpu.md", how_to_run_on_gpu);
+}
```

Comment:
I think it should be a module

like

```rust
#[cfg(featyre = "gpu")]
mod test_gpu_user_doc {
    doctest!(...);
}
```

the same for cpu but with the

#[cfg(not(feature = "gpu"))]

as the functions are not called, the doctest! macro creates the entry points

Reply:
You're right